src/components/StructureExplorer/MoleculeList.tsx

```diff
@@ -188,7 +192,14 @@ export function MoleculeList({
     return molecules;
   }, [idCode, molecules, db, search]);
 
+  const gridRef = useRef<Grid>(null);
+
+  const goToTop = () => {
+    gridRef.current?.scrollTo({ scrollTop: 0 });
+  };
+
   const molFiltered = useMemo(() => {
+    goToTop();
```

Comment:
⚠️ don't do side-effects in `useMemo`. I think we should use `useLayoutEffect` here.